com.ibm.streamsx.slack/com.ibm.streamsx.slack.service/SlackService.spl

```diff
@@ -0,0 +1,63 @@
+//
+// ****************************************************************************
+// * Copyright (C) 2017, International Business Machines Corporation          *
+// * All rights reserved.                                                     *
+// ****************************************************************************
+//
+
+/**
+ * A service which subscribes to a user-specified topic (in the parameters). 
+ * The incoming tuple should contain a jsonString. The jsonString, once converted, 
+ * should contain a message attribute with the text contents to send to Slack.
+ * 
+ * Input JSON Schema:
+ * {
+ *   message : string
+ * }
+ */
+
+namespace com.ibm.streamsx.slack.service;
+
+use com.ibm.streamsx.json::JSONToTuple ;
+use com.ibm.streamsx.slack::SendSlackMessage ;
+use com.ibm.streamsx.topology.topic::Subscribe ;
+
+public composite SlackService
+{
+	param
+		expression<rstring> $applicationConfigurationName ;
+		expression<rstring> $topic ;
+		
+	type
+		JSONType = rstring jsonString ;
+		
+	graph
+		/**
+		 * The Subscribe ingests data from the specified topic and outputs it.
+		 */
+		stream<JSONType> Subscribe_0 = Subscribe()
+		{
+			param
+				topic : $topic ;
+				streamType : JSONType ;
+		}
+		
+		/**
+		 * The JSONToTuple extracts the message attribute from the incoming JSON 
+		 * string and outputs it.
+		 */
+		stream<rstring message> JSONToTuple_0 = JSONToTuple(Subscribe_0)
```

Comment:
The way this is written, the *SendSlackMessage* operator can only read from the *message* attribute. I'm not sure if there's a way to allow users to specify a different attribute with a composite parameter (eg. $messageAttribute).

I remember trying to do this with another toolkit but not figuring out a solution.